macros/clone_schema.sql

```diff
@@ -0,0 +1,15 @@
+{% macro clone_schema(source_schema, destination_schema) %}
+  
+  {% if source_schema is not none and destination_schema is not none and source_schema != destination_schema %}
+    
+    {{ log("Cloning existing schema " ~ source_schema ~ " into schema " ~ destination_schema, info=True) }}
+    
+    {% call statement('clone_schema', fetch_result=True, auto_begin=False) -%}
+        CREATE OR REPLACE SCHEMA {{ destination_schema }} CLONE {{ source_schema }}
+    {%- endcall %}
+    
+    {%- set result = load_result('clone_schema') -%}
+    {{ log(result['data'][0][0], info=True)}}
+
```

Comment:
@jtalmi Thanks for contributing this!

I think this could be slightly improved by adding an _else_ condition. Right now I think nothing would happen if the arguments are not provided properly and the user might not have a clue on what's happening. Providing an helpful message through log() or exceptions.raise_compiler_error() would make this more user friendly.

What do you think?

I have no other comments otherwise and I'll be happy to merge this. :)

Reply:
great suggestion, i added an exception and simplified the conditional, including removing the part which forbade self-cloning...snowflake allows it so this macro should allow it, although i don't know why one would want to.

Reply:
Looks good! Thanks :)